src/wallet/scriptpubkeyman.cpp

```diff
@@ -278,6 +278,8 @@ bool LegacyScriptPubKeyMan::GetReservedDestination(const OutputType type, bool i
     if (!ReserveKeyFromKeyPool(index, keypool, internal)) {
         return false;
     }
+    LearnRelatedScripts(keypool.vchPubKey, type);
```

Comment:
3499674136167f662e5b0ef0b599c02f5a959558

You gave ACK in #17237 so I wonder I is this here?

Reply:
> [3499674](https://github.com/bitcoin/bitcoin/commit/3499674136167f662e5b0ef0b599c02f5a959558)
> 
> You gave ACK in #17237 so I wonder why is this here?

Better to ask this question in #17373 where this commit was moved than this PR. Specifically [17373/commits/35c37b5](https://github.com/bitcoin/bitcoin/pull/17373/commits/35c37b59c5bb918c15aca3092b6506ef3974e50c)

Probably this code change just precedes your PR

Reply:
It would be nice to get more review on #17373, even if this question won't be raised. The changes there are confusing to me, and might be more straightforward to you. (The changes there are minor and seem safe but I don't understand why they are made.)

Reply:
You mean #17373, not #17237 right? If so I'll review the first.